docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+# Rollout Strategy in Fleet
+
+Fleet uses a rollout strategy to control how apps are deployed across clusters. This feature allows you to define the order and grouping of cluster deployments, enabling controlled rollouts, and safer updates. Fleet evaluates the **Ready** status of each `BundleDeployment` to determine when to proceed to the next group. For more information, refer to [Status Fields](ref-status-fields.md).
+
+For initial deployments:
+
+* One cluster may be shown as **NotReady**.
+
+* Remaining clusters appear as **Pending**.
+
+For rollouts:
+
+* One cluster may be **NotReady**.
+
+* Remaining clusters are marked **OutOfSync** until the current one becomes ready
+
+Fleet creates up to 50 BundleDeployments at a time,  regardless of the rollout configuration. After creating a batch, Fleet evaluates the deployment status. If the current batch meets readiness criteria, Fleet proceeds to the next batch. Otherwise, it pauses the rollout.
+
+The rollout behavior is configured in the [`rolloutStrategy` in the `fleet.yaml`](ref-fleet-yaml.md)
+
+## How Does Partitioning Work?
+
+Partitions are considered non-ready if any cluster in the partition is not Ready. Deployment continues only if the number of non-ready partitions remains below maxUnavailablePartitions.
```

Comment:
> Partitions are considered non-ready if any cluster in the partition is not Ready.

The value of `maxUnavailable` that can be configured in manual partitioning (not the one in `rolloutStrategy.maxUnavailable` determines when a partition is being considered non-ready (or unavailable). If no value is set for that field when manually partitioning, or if automatic partitioning is used, then the value of `rolloutStrategy.maxUnavailable` is used to determine when a partition may be considered ready/available.